test/test_service.py

```diff
@@ -76,6 +76,16 @@ def _create_recommendations(self, count):
             test_recommendation.id = new_recommendation['id']
             recommendations.append(test_recommendation)
         return recommendations
+		
+    def test_get_recommendation(self):
+        """ Get a single Recommendation """
+        # get the id of a recommendation
+        test_rec = self._create_recommendations(1)[0]
+        resp = self.app.get('/recommendations/{}'.format(test_rec.id),
+                            content_type='application/json')
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        data = resp.get_json()
+        self.assertEqual(data['id'], test_rec.id)
 
     def test_create_recommendation(self):
         """ Create a new Recommendation """
```

Comment:
it seems like the problem was caused by the RecommendationFactory. If you called RecommendationFactory() then the id won't start from 0 but continues to increment. 
you can use     
  ```
test_recommendation = Recommendation(customer_id=2,
                                        product_id=3, recommend_product_id=4,
                                        recommend_type="upscale")
```
to avoid it and try not to use RecommendationFactory()

Reply:
to create an entry in the database, id must start from 0, or error would occurs

Reply:
I made an adjustment before reading this comment that seemed to have fixed the issue. I edited the deserialize model and test. Can you let me know if that change is acceptable? With that change I use the factory same as before and the test cases pass. If you think it is better to avoid the factory then I will go that route instead.

Reply:
Your change looks better to me, as we may need the factory method in later tests. Thanks for fixing this!